annif/openapi/annif.yaml

```diff
@@ -0,0 +1,274 @@
+openapi: 3.0.1
+info:
+  title: Annif REST API
+  version: v1
+servers:
+- url: /v1
+paths:
+  /projects:
+    get:
+      tags:
+      - Project administration
+      summary: get a list of projects
+      operationId: annif.rest.list_projects
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProjectList'
+  /projects/{project_id}:
+    get:
+      tags:
+      - Project administration
+      summary: show project information
+      operationId: annif.rest.show_project
+      parameters:
+      - $ref: '#/components/parameters/project_id'
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Project'
+        404:
+          description: Project not found
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/Problem'
+  /projects/{project_id}/suggest:
+    post:
+      tags:
+      - Automatic subject indexing
+      summary: suggest subjects for a given text
+      operationId: annif.rest.suggest
+      parameters:
+      - $ref: '#/components/parameters/project_id'
+      requestBody:
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              required:
+              - text
+              properties:
+                text:
+                  type: string
+                  description: input text
+                limit:
+                  type: integer
+                  description: maximum number of results to return
+                  format: int32
+                  default: 10
+                threshold:
+                  type: number
+                  description: minimum score threshold, below which results will not
+                    be returned
+                  default: 0.0
```

Comment:
The numerical values could have min. and max. specified. 

I remember at some point I could produce an error in Annif side when using negative value for some parameter, but cannot do it now anymore, so this is ok as it is.

Reply:
I added sensible minimum and maximum values, thanks for the suggestion!